test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -192,6 +261,1427 @@ public void testCreateDeleteKeyValueTable() {
+                versionList1.size(), entryList1.size());
+        log.info("Version list1: {}", versionList1);
+
+        List<Integer> keyList = new ArrayList<>();
+        for (int i = 0; i < keyArray.length; i++) {
+            keyList.add(keyArray[i]);
+        }
+        log.info("Key list: {}", keyList);
+        List<TableEntry<Integer, String>> getEntryList1 = newKeyValueTable.getAll(keyFamily, keyList).join();
+        assertNotNull("Get Entry List1 should not be empty", getEntryList1);
+        assertEquals("Get Entry List1 size should be same as get entry list1 size",
+                getEntryList1.size(), entryList1.size());
+        log.info("Get Entry List1: {}", getEntryList1);
+        for (int i = 0; i < keyArray.length; i++) {
+            assertEquals("Corresponding key for getEntryList should be as inserted",
+                    keyList.get(i), getEntryList1.get(i).getKey().getKey());
+            assertEquals("Corresponding key for entryList should be as inserted",
+                    entryList1.get(i).getKey().getKey(), getEntryList1.get(i).getKey().getKey());
+        }
+
+        // Comparing each & every entry values of both entrylist and getEntryList.
+        for (int i = 0; i < keyArray.length; i++) {
+            assertEquals("All string values of both entryList1 and getEntryList1 should be equal",
+                    entryList1.get(i).getValue(), getEntryList1.get(i).getValue());
+
+        }
+
+        List<TableKey<Integer>> tableKeyList = new ArrayList<>();
+        for (int i = 0; i < getEntryList1.size(); i++) {
+            tableKeyList.add(getEntryList1.get(i).getKey());
+        }
+        log.info("Table Key List: {}", tableKeyList);
+
+        assertNull("No error should be obtained during key remove",
+                newKeyValueTable.removeAll(keyFamily, tableKeyList).join());
+        log.info("All the requested table entries with specified key & its version are removed");
+
+        List<TableEntry<Integer, String>> getEntryList2 = newKeyValueTable.getAll(keyFamily, keyList).join();
+        log.info("Get entry list2: {}", getEntryList2);
+        for (int i = 0; i < getEntryList2.size(); i++) {
+            assertNull("All table entries fetched for deleted/not available keys should be null",
+                    getEntryList2.get(i));
+        }
+
+        AssertExtensions.assertSuppliedFutureThrows(
+                "The removeAll request must not succeed as the entries are already deleted",
+                () -> newKeyValueTable.removeAll(keyFamily, tableKeyList),
+                ex -> ex instanceof NoSuchKeyException
+        );
+        log.info("Found NoSuchKeyException on attempt to delete already removed key entries");
+        log.info("Successfully completed test Conditional delete Key with KeyFamily using its version");
+    }
+
+    /**
+     * Test case-50: Delete KVT and check TableEntries are not accessible
+     */
+    @Test
+    public void testDeleteKVTRetriveKVP() {
```

Comment:
This test is good, but you can simplify it . You don't need to test all the keys. Doing one is sufficient.

Reply:
Thanks Andrei. Sure, I'll simplify this test by checking the functioning of KVT with single entry instead of 10 entries.

Reply:
Still keeping the same 10 entries as wanted to test whether a KVTable with few entries in it can be deleted directly without any issue.